src/components/Pill/PillButtonEnhanced/PillButtonEnhanced.tsx

```diff
@@ -38,70 +41,24 @@ export const PillButtonEnhanced = React.forwardRef(
             additionalContentLabel,
             additionalContentTooltip,
             maxWidth,
+            isDisabled,
             ...rest
         }: Props<PriorityItemValue>,
         ref: React.Ref<HTMLElement>
     ) => {
         const isActive = !!content;
         const propsForBem = { isOpen, isActive };
 
-        const getButtonIconAndHandler = () => {
-            if (content) {
-                return {
-                    icon: (
-                        <Clear
-                            aria-label={clearLabel}
-                            viewBox="0 0 24 24"
-                            height="16px"
-                            width="16px"
-                        />
-                    ),
-                    onClick: (e: React.MouseEvent) => {
-                        e.stopPropagation();
-                        e.preventDefault();
-                        onClear();
-                    },
-                };
-            }
-
-            if (isOpen) {
-                return {
-                    icon: (
-                        <UpArrow
-                            aria-label={upArrowLabel}
-                            viewBox="0 0 24 24"
-                            height="16px"
-                            width="16px"
-                        />
-                    ),
-                    onClick: (e: React.MouseEvent) => {
-                        e.stopPropagation();
-                        e.preventDefault();
-                    },
-                };
-            }
-            return {
-                icon: (
-                    <DownArrow
-                        aria-label={downArrowLabel}
-                        viewBox="0 0 24 24"
-                        height="16px"
-                        width="16px"
-                    />
-                ),
-                onClick: (e) => {
-                    e.stopPropagation();
-                    e.preventDefault();
-                },
-            };
-        };
-
-        const { icon: buttonIcon, onClick: handleButtonClick } = getButtonIconAndHandler();
+        const buttonIcon = isOpen ? (
```

Comment:
I would say to choose one of them and rotate it with animation, just like it is being done in ExpandableText.

Reply:
💯